pkg/controller/deployment/util/deployment_util.go

```diff
@@ -466,12 +468,12 @@ func MaxSurge(deployment apps.Deployment) int32 {
 // GetProportion will estimate the proportion for the provided replica set using 1. the current size
 // of the parent deployment, 2. the replica count that needs be added on the replica sets of the
 // deployment, and 3. the total replicas added in the replica sets of the deployment so far.
-func GetProportion(rs *apps.ReplicaSet, d apps.Deployment, deploymentReplicasToAdd, deploymentReplicasAdded int32) int32 {
+func GetProportion(ctx context.Context, rs *apps.ReplicaSet, d apps.Deployment, deploymentReplicasToAdd, deploymentReplicasAdded int32) int32 {
```

Comment:
The entire callchain looks like a simple getter, with no remote interaction. Therefore passing a logger is better suited to the purpose.

Use `klog.Logger` to avoid an extra import of `logr`.